src/tests/app.test.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { render /* screen, waitFor */ } from "@testing-library/react";
+// import axios from "axios";
+import App from "../components/app";
+
+describe("App", () => {
+  it("renders app components correctly", async () => {
+    // const mockedData = {
```

Comment:
Snapshot testing is about 'taking a picture' of the app when it first loads, to make sure the component framework is correct, rather than the entire app is loaded after things like api calls populate it with data. If you look at the snapshot as it is, thats exactly what its doing. If you think about it, the idea in testing in modular pieces like this is to test each individual part on its own merits, and all the things that would be rendered once the api call has completed are not handled by this component so we dont need to worry about testing them here

You can do this with a bit of extra manipulation, but you dont need to. Just asserting the snapshot is correct for the raw App component and the elements you expect to be there are there is enough


In the meantime, leaving loads of commented out code in here like this isnt a good idea